Color contrast survey updates (#1961)
act-rules/act-rules.github.io: _rules/text-contrast-afw4f7.md

This rule applies to any [visible][] character in a [text node][] that is a [child][] in the [flat tree][] of an [HTML element][], except if the [text node][] has an [ancestor][] in the [flat tree][] for which at least one of the following is true:

- **disabled widget**: the ancestor is a [inheriting semantic][] `widget` that is [disabled][]; or
- **disabled label**: the ancestor is used in the [accessible name][] of a [inheriting semantic][] `widget` that is [disabled][]; or

> - **disabled group**: the ancestor has a [semantic role][] of `group` and is [disabled][].

Reviewer: Also realising that the `disabled` bit is now part of all three conditions and could be refactored:

```markdown
This rule applies to any [visible][] character in a [text node][] that is a [child][] in the [flat tree][] of an [HTML element][], except if the [text node][] has an [ancestor][] in the [flat tree][] which is [disabled][] and for which at least one of the following is true:

- **disabled widget**: the ancestor is a [inheriting semantic][] `widget`; or
- **disabled label**: the ancestor is used in the [accessible name][] of a [inheriting semantic][] `widget`; or
- **disabled group**: the ancestor has a [semantic role][] of `group`.
```
Maybe the conditions would then need to be renamed simply `**widget**`, `**label**` and `**group**` 🤔 

Same can be done for the other rule.

We could also the widget/group conditions that are very similar ("the ancestor is a [inheriting semantic][] `group` or `widget`")

This is a bit unrelated, definitely not blocking this PR and can be tackled later if wanted.

This light gray text has a contrast ratio between 13:1 and 5:1 on the background image.

```html
<style>
p {
	color: #CCC;
	height: 50px;
	padding-top: 15px;
	background: #000 no-repeat -20px -20px url('/test-assets/contrast/black-hole.jpeg');

> 	text-shadow: 0px 0px 2px black;

Reviewer 2: I tested this locally. This looks good to me.

</script>
```

#### Passed Example 10

> This text is part of a widget because it is a child of a `button` element. The text has the default
> browser button text color on the default browser button background color. By default, this is black text on a
> light gray background with a contrast ratio of 18.26:1

Reviewer: I don't think we need to mention that the text is part of a widget since the rule doesn't care.
```suggestion
This text is using the default user agent colors for button, a black (`#000`) text with light gray (<insert value>) background resulting in a contrast ratio of 18.26:1
```
(I also couldn't find the actual colors in [Rendering](https://html.spec.whatwg.org/multipage/rendering.html) in the HTML living standard; this is a non-normative part.)

```html
<button>My button!</button>
```

#### Passed Example 11

> This text is part of a widget because it is a child of an element with the `role` attribute set to `button`.

Reviewer: Same, here, I don't think we need to mention that the text is part of a widget.

#### Failed Example 8

> The first `p` element has a contrast ratio of 12.6:1. The second `p` element, which contains an example of the Helvetica font, has a contrast ratio of 3.85:1. Because this provides information, and not only for aesthetic purposes, this is not considered [purely decorative][].

Reviewer: ```suggestion
The first `p` element has a contrast ratio of 12.6:1. The second `p` element, which contains an example of the Helvetica font, has a contrast ratio of 3.85:1. Because this provides information, and is not only for aesthetic purposes, this is not considered [purely decorative][].
```

</p>
```

#### Failed Example 9

> This text is part of a widget because it is a child of a `button` element. The button text has a contrast
> ratio of 3.85:1.

Reviewer: ```suggestion
This text in a button has a contrast ratio of 3.85:1.
```

```html
<button style="color: #777; background: #EEE;">My button!</button>
```

#### Failed Example 10

> This text is part of a widget because it is a child of an element with the `role` attribute set to `button`.
> The button text has a contrast ratio of 3.85:1.

Reviewer: ```suggestion
This text in a semantic button has a contrast ratio of 3.85:1.
```

#### Inapplicable Example 6

> This text is part of a widget because it is a child of a `button` element.

Reviewer 2: It would be good to have a disable button example in here. Maybe one with the native `disabed` attribute, and an ARIA one with `aria-disabled=true`?

Author: @WilcoFiers Added Inapplicable 10 and 11. Is this what you were expecting?

</div>
```

#### Inapplicable Example 10

> This text is part of a [disabled][] widget because it is a child of a `button` element with the `disabled` attribute.

Reviewer 2: ```suggestion
This text is part of a [disabled][] widget because it is a child of a `button` element with the `disabled` attribute.

```
